### PR TITLE
Create Main Page and add UI interactions

### LINKED ISSUE #18: Add navigation drawer component
This should show a list of groups and actions related to groups at the bottom when you touch the menu icon in the navbar

### LINKED ISSUE #19: Add group list item component
Should be similar to the client, this will display the group's picture to the left with the group's title and number of members to the right

### LINKED ISSUE #20: Create image grid component
Should be a grid of images belonging to the currently selected group (similar to the client)

### PR DESCRIPTION
**Pull request by suneettipirneni**
Adds the main page to the app, in addition the app now uses a unified navigation stack.

https://user-images.githubusercontent.com/77477100/114637542-a2c8de80-9c97-11eb-8249-992625ba24b9.mov

closes #18
closes #19
closes #20

### CODE REVIEW

File: src/components/main/ListItem.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      </View>
+    );
+  }
+
+  return (
+    <View style={styles.container}>
+      {renderIcon()}
+      <View style={styles.textContainer}>
+        <Text numberOfLines={1} style={[styles.title, { marginBottom: 5 }]}>{title}</Text>
+        <Text style={[styles.subtitle, { color: 'gray' }]}>{`${members} members`}</Text>
```
Comment (ctcuff) [inline src/components/main/ListItem.jsx:30]: Just a nitpick but this causes groups with one member to show as `1 members`

File: src/pages/Navigation.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <Drawer.Navigator initialRouteName="Home" drawerContent={GroupDrawer}>
+      <Drawer.Screen
+        name="Home"
+        component={MainPage}
+        options={({ navigation }) => ({
+          headerShown: true,
+          title: 'ImageUs',
+          headerLeft: () => (<Ionicons name="ios-reorder-three-sharp" size={36} onPress={() => navigation.toggleDrawer()} />),
+        })}
```
Comment (ctcuff) [inline src/pages/Navigation.jsx:22]: I think the icon needs a few px of left margin, it seems a little too close to the edge of the screen. Tested out `marginLeft: 16` and it seems to fit.

**Review by ctcuff — APPROVED** [review]
Awesome, LGTM, just a few small opinions about styles.

**Comment by suneettipirneni** [comment]
> Awesome, LGTM, just a few small opinions about styles.

Changes applied!

**Pull request merged**